src/v/cluster/types.h

```diff
@@ -953,14 +953,20 @@ struct topic_result
     explicit topic_result(model::topic_namespace t, errc ec = errc::success)
       : tp_ns(std::move(t))
       , ec(ec) {}
+    topic_result(model::topic_namespace t, errc ec, std::string_view msg)
+      : tp_ns(std::move(t))
+      , ec(ec)
+      , error_message{
+          ssx::sformat("{}: {}", make_error_code(ec).message(), msg)} {}
     model::topic_namespace tp_ns;
-    errc ec;
+    errc ec{errc::success};
+    std::optional<ss::sstring> error_message;
```

Comment:
I've seen people bump the serde::version when adding fields, but I'm sure that's not a requirement here.

Reply:
Did it anyway.